.github/workflows/database-backup.yml

```diff
@@ -4,72 +4,69 @@ concurrency: build_and_deploy_main
     - uses: actions/checkout@v4
       name: Checkout
 
-    - uses: azure/login@v2
-      with:
-        creds: ${{ secrets.AZURE_CREDENTIALS }}
-
-    - name: Setup postgres client
-      uses: DFE-Digital/github-actions/install-postgres-client@master
-      with:
-        version: 14
-
-    - name: Install kubectl
-      uses: azure/setup-kubectl@v4
-      with:
-        version: "v1.26.1" # default is latest stable
-
-    - uses: DFE-Digital/github-actions/set-kubelogin-environment@master
-      with:
-        azure-credentials: ${{ secrets.AZURE_CREDENTIALS }}
-
-    - name: K8 setup
-      shell: bash
-      run: |
-        make ci production get-cluster-credentials
-        make bin/konduit.sh
-
-    - name: Set environment variable
-      run: echo "BACKUP_FILE_NAME=ittm_prod_$(date +"%F")" >> $GITHUB_ENV
-
-    - name: Backup Prod DB
+    - name: Set environment variables
       run: |
-        bin/konduit.sh -t 7200 itt-mentor-services-production -- pg_dump -E utf8 --clean --if-exists --no-owner --verbose --no-password -f ${BACKUP_FILE_NAME}.sql
-        tar -cvzf ${BACKUP_FILE_NAME}.tar.gz ${BACKUP_FILE_NAME}.sql
+        source global_config/${{ env.DEPLOY_ENV }}.sh
+        tf_vars_file=${{ env.TF_VARS_PATH }}/${{ env.DEPLOY_ENV }}.tfvars.json
+        echo "CLUSTER=$(jq -r '.cluster' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "RESOURCE_GROUP_NAME=${AZURE_RESOURCE_PREFIX}-${SERVICE_SHORT}-${CONFIG_SHORT}-rg" >> $GITHUB_ENV
+        echo "STORAGE_ACCOUNT_NAME=${AZURE_RESOURCE_PREFIX}${SERVICE_SHORT}dbbkp${CONFIG_SHORT}sa" >> $GITHUB_ENV
+        TODAY=$(date +"%F")
+        echo "BACKUP_FILE=${SERVICE_SHORT}_${CONFIG_SHORT}_${TODAY}.sql" >> $GITHUB_ENV
+        echo "DB=${AZURE_RESOURCE_PREFIX}-${SERVICE_SHORT}-${CONFIG_SHORT}-pg" >> $GITHUB_ENV
 
-    - name: Set up environment variables
-      shell: bash
+    - name: Set PTR variables
+      if: ${{ github.event.inputs.PTRdatabase == 'true' }}
       run: |
-        echo "STORAGE_ACCOUNT_RG=s189p01-ittms-pd-rg" >> $GITHUB_ENV
-        echo "STORAGE_ACCOUNT_NAME=s189p01ittmsdbbkppdsa" >> $GITHUB_ENV
+        echo "PTR_DB=${{ env.DB }}-PTR" >> $GITHUB_ENV
 
-    - name: Set Connection String
-      run: |
-        STORAGE_CONN_STR=$(az storage account show-connection-string -g  $STORAGE_ACCOUNT_RG -n $STORAGE_ACCOUNT_NAME --query 'connectionString')
-        echo "::add-mask::$STORAGE_CONN_STR"
-        echo "AZURE_STORAGE_CONNECTION_STRING=$STORAGE_CONN_STR" >> $GITHUB_ENV
-
-    - name: Upload Backup to Azure Storage
-      run: |
-        az config set extension.use_dynamic_install=yes_without_prompt
-        az config set core.only_show_errors=true
-        az storage azcopy blob upload --container database-backup \
-        --source ${BACKUP_FILE_NAME}.tar.gz
+    - name: Backup ${{ env.DEPLOY_ENV }} postgres
+      if: ${{ github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.overwriteThisMorningsBackup == 'true' ) }}
```

Comment:
Is this required at the step level? It's already set at job level

Reply:
removed from lower level